atlasdb-commons/src/test/java/com/palantir/common/compression/LZ4CompressionTests.java

```diff
@@ -0,0 +1,155 @@
+/**
+ * Copyright 2016 Palantir Technologies
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.common.compression;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Arrays;
+import java.util.Random;
+
+import org.junit.After;
+import org.junit.Test;
+
+import com.google.common.io.ByteStreams;
+
+import net.jpountz.lz4.LZ4BlockInputStream;
+
+public class LZ4CompressionTests {
+
+    private static final byte SINGLE_VALUE = 42;
+    private static final int BLOCK_SIZE = 1 << 16; // 64 KB
+
+    private ByteArrayInputStream uncompressedStream;
+    private LZ4CompressingInputStream compressingStream;
+    private LZ4BlockInputStream decompressingStream;
+
+    @After
+    public void after() throws IOException {
+        uncompressedStream.close();
+        compressingStream.close();
+        decompressingStream.close();
+    }
+
+    @Test
+    public void testEmptyStream() throws IOException {
+        initializeStreams(new byte[0]);
+        assertStreamIsEmpty(decompressingStream);
+    }
+
+    @Test
+    public void testSingleCharacterStream() throws IOException {
+        byte[] uncompressedData = new byte[] { SINGLE_VALUE };
+        initializeStreams(uncompressedData);
+        verifyStreamContents(uncompressedData);
+    }
+
+    @Test
+    public void testSingleCharacterSteam_singleByteRead() throws IOException {
+        byte[] uncompressedData = new byte[] { SINGLE_VALUE };
+        initializeStreams(uncompressedData);
+        int value = decompressingStream.read();
+
+        assertEquals(uncompressedData[0], value);
+        assertStreamIsEmpty(decompressingStream);
+    }
+
+    @Test
+    public void testSingleBlock() throws IOException {
```

Comment:
Should probably be `testSingleBlock_compressible`, to match with the below.

Also, while in general the extracted methods are very nice, you do have duplicated test code (this and the below method both have pairs which only differ in the size of the byte array). Consider extracting another two helper methods, one for compressible and one for incompressible.